### DISCUSSION TITLE
Getting some error while setting up Mastodon

### DISCUSSION THREAD

**Discussion by mitexleo** [mastodon/mastodon#20320] [General]
Hi 👋, 
I'm facing a problem which I don't understand 😕..
While trying to precompiled assets I'm getting following error ..

<--- Last few GCs --->

[73650:0x57cbe50]    45530 ms: Mark-sweep (reduce) 480.2 (493.8) -> 479.5 (494.3) MB, 473.5 / 0.0 ms  (average mu = 0.184, current mu = 0.037) allocation failure scavenge might not succeed
[73650:0x57cbe50]    46038 ms: Mark-sweep (reduce) 480.5 (494.3) -> 479.9 (494.8) MB, 504.3 / 0.0 ms  (average mu = 0.104, current mu = 0.007) allocation failure scavenge might not succeed

<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 0xb06730 node::Abort() [node]
 2: 0xa1b6d0  [node]
 3: 0xce1e60 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
 4: 0xce2207 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
 5: 0xe99875  [node]
 6: 0xea953d v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
 7: 0xeac23e v8::internal::Heap::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
 8: 0xe6d77a v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationType, v8::internal::AllocationOrigin) [node]
 9: 0x11e64e6 v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [node]
10: 0x15da159  [node]

![SmartSelect_20221110-171255_Termux](https://user-images.githubusercontent.com/114842102/201077252-de88fe45-2f5e-44af-be07-83dae4da30c4.jpg)

Note :

* I increased the memory to 5 GB (total) buy swaping storage.
* I also increased the Node Memory Limit to 4Gb by following this (https://sebhastian.com/javascript-heap-out-of-memory/) guide ..

TIA

**Comment by mitexleo** [comment]
Another Question: What should be the File Permission for the Swap File ? (Default: 644)

**Comment by prfiredragon** [comment]
Same error here. After RAIL_ENV='production' add NODE_OPTIONS='--max-old-space-size=8192'
